amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/StreamIdentifier.java

```diff
@@ -101,129 +84,82 @@ public String toString() {
-     */
-    public static StreamIdentifier multiStreamInstance(String serializationOrArn) {
-        return multiStreamInstance(serializationOrArn, null);
-    }
-
-    /**
-     * Create a multi stream instance for StreamIdentifier from serialized stream identifier.
-     *
-     * @param serializationOrArn serialized {@link StreamIdentifier} or AWS ARN of a Kinesis stream
-     * @param kinesisRegion Kinesis client endpoint, and also where the stream(s) to be
-     *          processed are located. A null will default to the caller's region.
+     * Create a multi stream instance for StreamIdentifier from serialized stream identifier
+     * of format {@link #STREAM_IDENTIFIER_PATTERN}
      *
-     * @see #multiStreamInstance(String)
-     * @see #serialize()
+     * @param streamIdentifierSer
+     * @return StreamIdentifier
      */
-    public static StreamIdentifier multiStreamInstance(String serializationOrArn, Region kinesisRegion) {
-        final StreamIdentifier fromSerialization = fromSerialization(serializationOrArn, kinesisRegion);
-        if (fromSerialization != null) {
-            return fromSerialization;
-        }
-        final StreamIdentifier fromArn = fromArn(serializationOrArn, kinesisRegion);
-        if (fromArn != null) {
-            return fromArn;
+    public static StreamIdentifier multiStreamInstance(String streamIdentifierSer) {
+        final Matcher matcher = STREAM_IDENTIFIER_PATTERN.matcher(streamIdentifierSer);
+        if (matcher.matches()) {
+            final String accountId = matcher.group("accountId");
+            final String streamName = matcher.group("streamName");
+            final Long creationEpoch = Long.valueOf(matcher.group("creationEpoch"));
+
+            return StreamIdentifier.builder()
+                    .accountIdOptional(Optional.of(accountId))
+                    .streamName(streamName)
+                    .streamCreationEpochOptional(Optional.of(creationEpoch))
+                    .build();
         }
 
-        throw new IllegalArgumentException("Unable to deserialize StreamIdentifier from " + serializationOrArn);
+        throw new IllegalArgumentException("Unable to deserialize StreamIdentifier from " + streamIdentifierSer);
     }
 
     /**
-     * Create a single stream instance for StreamIdentifier from stream name.
+     * Create a multi stream instance for StreamIdentifier from stream {@link Arn}
+     * of format {@link #STREAM_ARN_PATTERN} and creation epoch.
      *
-     * @param streamNameOrArn stream name or AWS ARN of a Kinesis stream
-     *
-     * @see #singleStreamInstance(String, Region)
+     * @param streamARN
+     * @param creationEpoch
+     * @return StreamIdentifier
      */
-    public static StreamIdentifier singleStreamInstance(String streamNameOrArn) {
-        return singleStreamInstance(streamNameOrArn, null);
+    public static StreamIdentifier multiStreamInstance(Arn streamARN, Long creationEpoch) {
+        validateARN(streamARN);
```

Comment:
1. It might benefit to validate `creationEpoch > 0`. Negative epochs make no sense.
1. This is a trade-off that allows customers to self-harm: if they don't provide the correct epoch, and KCL doesn't validate it against the _actual_ stream, customers might recycle the epoch for two editions of the same stream which might present odd/deleterious behavior in lease management. (e.g., create stream `A` at epoch `t1`, create `StreamIdentifer(A, t1)`, delete `A`, create stream `A` at epoch `t2`, reuse old epoch to create `StreamIdentifier(A, t1)`)

... or, potentially, a customer decides the epoch correctness doesn't matter and they provide `0` (or 1, or 42, or ...).

At the least, **this expected behavior must be very clearly detailed in the Javadoc**.